In [1]:
!jupyter nbconvert Titanic.ipynb --to html --no-input --no-prompt --output 09__notebook_as_report.html

[NbConvertApp] Converting notebook Titanic.ipynb to html
[NbConvertApp] WARNING | Alternative text is missing on 6 image(s).
[NbConvertApp] Writing 539220 bytes to 09__notebook_as_report.html


## O Danych

Dane o pasażerach Titanica

Zbiór danych zawiera informacje o pasażerach RMS Titanic, który zatonął 15 kwietnia 1912 roku po zderzeniu z górą
lodową. Dane obejmują takie atrybuty jak klasa podróży, wiek, płeć, liczba rodzeństwa/małżonków na pokładzie,
liczba rodziców/dzieci na pokładzie, cena biletu oraz miejsce zaokrętowania.

Zbiór zawiera także informację o tym, czy pasażer przeżył katastrofę.

Titanic przewoził ponad 2,200 osób, z czego ponad 1,500 zginęło, co czyni tę katastrofę jedną z najbardziej
tragicznych w historii morskiej.

Kolumny:

* **pclass** - Klasa biletu
* **survived** - Czy pasażer przeżył katastrofę
* **name** - Imię i nazwisko pasażera
* **sex** - Płeć pasażera
* **age** - Wiek pasażera
* **sibsp** - Liczba rodzeństwa/małżonków na pokładzie
* **parch** - Liczba rodziców/dzieci na pokładzie
* **ticket** - Numer biletu
* **fare** - Cena biletu
* **cabin** - Numer kabiny
* **embarked** - Port, w którym pasażer wszedł na pokład (C = Cherbourg, Q = Queenstown, S = Southampton)
* **boat** - Numer łodzi ratunkowej
* **body** - Numer ciała (jeśli pasażer nie przeżył i ciało zostało odnalezione)
* **home.dest** - Miejsce docelowe

![RMS Titanic](https://preview.redd.it/70hdhge2b2951.png?width=640&crop=smart&auto=webp&s=44d8a0ce0e54e0a64ef71129e1778f805c11e011)

![RMS Titanic](https://scontent-waw2-1.xx.fbcdn.net/v/t39.30808-6/341070359_605768921461890_7496459028713192087_n.jpg?_nc_cat=109&ccb=1-7&_nc_sid=6ee11a&_nc_ohc=B1eUpUEk1_kQ7kNvwGrwo0M&_nc_oc=AdmN_d34i5rj7oMCVEJtRnW6iYIDi-95cckL19l_2TiTTSxsJ5yAYPmif6K-CRvr_4J4lXXfXwMDlWgz8N80cgiH&_nc_zt=23&_nc_ht=scontent-waw2-1.xx&_nc_gid=RU2txMs2V3Dj23Hc925Kow&oh=00_AfTdZmWerg8IUlHszxFwVauVNbKr2P6qgupq6uZc8uWr0A&oe=6886713E)

![RMS Titanic](https://blogger.googleusercontent.com/img/b/R29vZ2xl/AVvXsEi_zYnhos67_KoB1jQTXmno84co-YAzxpo90Cg-3kEs1MN5RNBWpN8cPh3lmphqevmHDjnIUhu5NlMPvkDQDhEZKfdafhza4zhveOHZbQrX6Xy9b4l7VpisfwHmHyIgDpBRJqjiJ0NnFOg/s1600/Titanic_breaks_in_half.jpg)

![map](https://titanicfacts.net/wp-content/uploads/2018/06/titanic-maiden-voyage-route.gif)

# 1. Analiza wstepna

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('26__titanic.csv', sep=",")
df

pclass  survived                                             name  \
0        1.0       1.0                    Allen, Miss. Elisabeth Walton   
1        1.0       1.0                   Allison, Master. Hudson Trevor   
2        1.0       0.0                     Allison, Miss. Helen Loraine   
3        1.0       0.0             Allison, Mr. Hudson Joshua Creighton   
4        1.0       0.0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1305     3.0       0.0                            Zabour, Miss. Thamine   
1306     3.0       0.0                        Zakarian, Mr. Mapriededer   
1307     3.0       0.0                              Zakarian, Mr. Ortin   
1308     3.0       0.0                               Zimmerman, Mr. Leo   
1309     NaN       NaN                                              NaN   

         sex      age  sibsp  parch  ticket      fare    cabin embarked boat  \
0     female  29.0000    0.0    0.0   24160  211.3375       B5        S    2   
1       male   0.9167    1.0    2.0  113781  151.5500  C22 C26        S   11   
2     female   2.0000    1.0    2.0  113781  151.5500  C22 C26        S  NaN   
3       male  30.0000    1.0    2.0  113781  151.5500  C22 C26        S  NaN   
4     female  25.0000    1.0    2.0  113781  151.5500  C22 C26        S  NaN   
...      ...      ...    ...    ...     ...       ...      ...      ...  ...   
1305  female      NaN    1.0    0.0    2665   14.4542      NaN        C  NaN   
1306    male  26.5000    0.0    0.0    2656    7.2250      NaN        C  NaN   
1307    male  27.0000    0.0    0.0    2670    7.2250      NaN        C  NaN   
1308    male  29.0000    0.0    0.0  315082    7.8750      NaN        S  NaN   
1309     NaN      NaN    NaN    NaN     NaN       NaN      NaN      NaN  NaN   

       body                        home.dest  
0       NaN                     St Louis, MO  
1       NaN  Montreal, PQ / Chesterville, ON  
2       NaN  Montreal, PQ / Chesterville, ON  
3     135.0  Montreal, PQ / Chesterville, ON  
4       NaN  Montreal, PQ / Chesterville, ON  
...     ...                              ...  
1305    NaN                              NaN  
1306  304.0                              NaN  
1307    NaN                              NaN  
1308    NaN                              NaN  
1309    NaN                              NaN  

[1310 rows x 14 columns]

### Wgrajmy kilka razy 10 losowych rzedow w celu dokladniejszych obserwacji:

In [4]:
df.sample (10) 

pclass  survived                                               name  \
1087     3.0       0.0                   Olsson, Mr. Nils Johan Goransson   
586      2.0       1.0      Weisz, Mrs. Leopold (Mathilde Francoise Pede)   
547      2.0       0.0                                 Richard, Mr. Emile   
1011     3.0       0.0                McNamee, Mrs. Neal (Eileen O'Leary)   
673      3.0       0.0                                Betros, Mr. Tannous   
352      2.0       1.0  Brown, Mrs. Thomas William Solomon (Elizabeth ...   
802      3.0       0.0                                    Flynn, Mr. John   
591      2.0       1.0                       West, Miss. Constance Mirium   
565      2.0       0.0                     Sobey, Mr. Samuel James Hayden   
1062     3.0       0.0                      Nilsson, Mr. August Ferdinand   

         sex   age  sibsp  parch         ticket     fare cabin embarked boat  \
1087    male  28.0    0.0    0.0         347464   7.8542   NaN        S  NaN   
586   female  29.0    1.0    0.0         228414  26.0000   NaN        S   10   
547     male  23.0    0.0    0.0  SC/PARIS 2133  15.0458   NaN        C  NaN   
1011  female  19.0    1.0    0.0         376566  16.1000   NaN        S  NaN   
673     male  20.0    0.0    0.0           2648   4.0125   NaN        C  NaN   
352   female  40.0    1.0    1.0          29750  39.0000   NaN        S   14   
802     male   NaN    0.0    0.0         368323   6.9500   NaN        Q  NaN   
591   female   5.0    1.0    2.0     C.A. 34651  27.7500   NaN        S   10   
565     male  25.0    0.0    0.0     C.A. 29178  13.0000   NaN        S  NaN   
1062    male  21.0    0.0    0.0         350410   7.8542   NaN        S  NaN   

      body                              home.dest  
1087   NaN                                    NaN  
586    NaN     Bromsgrove, England / Montreal, PQ  
547    NaN                   Paris / Montreal, PQ  
1011  53.0                                    NaN  
673    NaN                                  Syria  
352    NaN  Cape Town, South Africa / Seattle, WA  
802    NaN                                    NaN  
591    NaN                    Bournmouth, England  
565    NaN                Cornwall / Houghton, MI  
1062   NaN                                    NaN

### Wstępna analiza pokazuje, ze RMS Titanic podróżowało 1310 pasażerów. Skoro na statku było 2200 osób, zakładam ze reszta tj 890 osób to zaloga. Te dane przydadzą się przy późniejszej analizie przeżywalności. Mimo ze wiemy, ze Titanic płynął przez Atlantyk do Nowego Jorku, pojawia się duża liczba miejsc docelowych, w tym Paryż i Londyn, 238 osób, w tym 1 klasa nie ma miejsca docelowego. Zastanawiam się, czy nie jest to błąd w pomiarze danych. Jest tez ogromna rozbieżność w cenach biletów. Jest 1 tajemniczy pasażer (1309) o którym nic nie wiadomo, same NaN. Pasazer 236 nie ma miejsca docelowego, mimo ze 1 klasa. Na statku były 3 klasy pasażerów, 186 kajut było rozdysponowanych pomiędzy pasażerów 1 i 2 klasy.

In [5]:
df.nunique()

pclass          3
survived        2
name         1307
sex             2
age            98
sibsp           7
parch           8
ticket        929
fare          281
cabin         186
embarked        3
boat           27
body          121
home.dest     369
dtype: int64

### Zwraca uwage mala liczba 27 szalup ratunkowych. 186 kajut. 121 cial odnaleziono(na 1500). Ceny biletów wykazują aż 281 unikatowych wartości! Ciekawie byloby sprawdzić od czego zależy tak duża rozbieżność.

In [6]:
df.describe()

pclass     survived          age        sibsp        parch  \
count  1309.000000  1309.000000  1046.000000  1309.000000  1309.000000   
mean      2.294882     0.381971    29.881135     0.498854     0.385027   
std       0.837836     0.486055    14.413500     1.041658     0.865560   
min       1.000000     0.000000     0.166700     0.000000     0.000000   
25%       2.000000     0.000000    21.000000     0.000000     0.000000   
50%       3.000000     0.000000    28.000000     0.000000     0.000000   
75%       3.000000     1.000000    39.000000     1.000000     0.000000   
max       3.000000     1.000000    80.000000     8.000000     9.000000   

              fare        body  
count  1308.000000  121.000000  
mean     33.295479  160.809917  
std      51.758668   97.696922  
min       0.000000    1.000000  
25%       7.895800   72.000000  
50%      14.454200  155.000000  
75%      31.275000  256.000000  
max     512.329200  328.000000

### Ogromna rozbieżność w cenach biletów od 0 czy 7 (funtów, dolarow?) do 512. Najmłodszy pasażer mial  1 miesiąc, a najstarszy 80 lat.

In [7]:
df.info ()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1310 entries, 0 to 1309
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   pclass     1309 non-null   float64
 1   survived   1309 non-null   float64
 2   name       1309 non-null   object 
 3   sex        1309 non-null   object 
 4   age        1046 non-null   float64
 5   sibsp      1309 non-null   float64
 6   parch      1309 non-null   float64
 7   ticket     1309 non-null   object 
 8   fare       1308 non-null   float64
 9   cabin      295 non-null    object 
 10  embarked   1307 non-null   object 
 11  boat       486 non-null    object 
 12  body       121 non-null    float64
 13  home.dest  745 non-null    object 
dtypes: float64(7), object(7)
memory usage: 143.4+ KB


### W szalupach ratunkowych bylo lacznie 486 osob

In [8]:
df['survived'].value_counts()

survived
0.0    809
1.0    500
Name: count, dtype: int64

### 500 pasazerow przezylo katastrofe, czyli 14 pasazerow nie bylo w szalupie ratunkowej i przezylo. Skoro odratowano 700 osób, w tym 500 pasażerów, przyjąć należy ze odratowano tez 200 członków załogi.

In [9]:
df['home.dest'].value_counts()

home.dest
New York, NY                                    64
London                                          14
Montreal, PQ                                    10
Paris, France                                    9
Cornwall / Akron, OH                             9
                                                ..
Chelsea, London                                  1
Harrow-on-the-Hill, Middlesex                    1
Copenhagen, Denmark                              1
Guernsey / Montclair, NJ and/or Toledo, Ohio     1
Antwerp, Belgium / Stanton, OH                   1
Name: count, Length: 369, dtype: int64

### Uwage zwracaja miejsca docelowe Londyn, Paryz i inne miasta europejskie. Sa to prawdopodobnie pasazerowie nie podrozujacy aby osiedlic sie w Ameryce, lecz plynacy na wycieczke i powracajacy pozniej do domu w Europie. Na taka podroz moglaby sobie pozwolic tylko 1 klasa, ewentualnie 2. Sprawdzmy te teze:

In [10]:
Paryzanie = df[df['home.dest'].str.contains('Paris, France', na=False)]
Paryzanie_klasa = Paryzanie['pclass']
result = {
    "type": "dataframe",
    "value": Paryzanie_klasa}
Paryzanie

pclass  survived                                            name     sex  \
12      1.0       1.0                   Aubart, Mme. Leontine Pauline  female   
68      1.0       1.0                        Chevre, Mr. Paul Romaine    male   
133     1.0       1.0                        Goldenberg, Mr. Samuel L    male   
134     1.0       1.0    Goldenberg, Mrs. Samuel L (Edwiga Grabowska)  female   
169     1.0       0.0                      Isham, Miss. Ann Elizabeth  female   
187     1.0       1.0                       Lines, Miss. Mary Conover  female   
188     1.0       1.0  Lines, Mrs. Ernest H (Elizabeth Lindsey James)  female   
196     1.0       1.0                            Marechal, Mr. Pierre    male   
219     1.0       1.0                       Omont, Mr. Alfred Fernand    male   
242     1.0       1.0                   Rosenbaum, Miss. Edith Louise  female   
276     1.0       0.0                   Spencer, Mr. William Augustus    male   
277     1.0       1.0  Spencer, Mrs. William Augustus (Marie Eugenie)  female   

      age  sibsp  parch      ticket      fare cabin embarked boat  body  \
12   24.0    0.0    0.0    PC 17477   69.3000   B35        C    9   NaN   
68   45.0    0.0    0.0    PC 17594   29.7000    A9        C    7   NaN   
133  49.0    1.0    0.0       17453   89.1042   C92        C    5   NaN   
134   NaN    1.0    0.0       17453   89.1042   C92        C    5   NaN   
169  50.0    0.0    0.0    PC 17595   28.7125   C49        C  NaN   NaN   
187  16.0    0.0    1.0    PC 17592   39.4000   D28        S    9   NaN   
188  51.0    0.0    1.0    PC 17592   39.4000   D28        S    9   NaN   
196   NaN    0.0    0.0       11774   29.7000   C47        C    7   NaN   
219   NaN    0.0    0.0  F.C. 12998   25.7417   NaN        C    7   NaN   
242  33.0    0.0    0.0    PC 17613   27.7208   A11        C   11   NaN   
276  57.0    1.0    0.0    PC 17569  146.5208   B78        C  NaN   NaN   
277   NaN    1.0    0.0    PC 17569  146.5208   B78        C    6   NaN   

                        home.dest  
12                  Paris, France  
68                  Paris, France  
133  Paris, France / New York, NY  
134  Paris, France / New York, NY  
169    Paris, France New York, NY  
187                 Paris, France  
188                 Paris, France  
196                 Paris, France  
219                 Paris, France  
242                 Paris, France  
276                 Paris, France  
277                 Paris, France

### Teza potwierdzila sie, wszyscy Paryzanie to pasazerowie 1 klasy

In [11]:
df.head (10)

pclass  survived                                             name     sex  \
0     1.0       1.0                    Allen, Miss. Elisabeth Walton  female   
1     1.0       1.0                   Allison, Master. Hudson Trevor    male   
2     1.0       0.0                     Allison, Miss. Helen Loraine  female   
3     1.0       0.0             Allison, Mr. Hudson Joshua Creighton    male   
4     1.0       0.0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   
5     1.0       1.0                              Anderson, Mr. Harry    male   
6     1.0       1.0                Andrews, Miss. Kornelia Theodosia  female   
7     1.0       0.0                           Andrews, Mr. Thomas Jr    male   
8     1.0       1.0    Appleton, Mrs. Edward Dale (Charlotte Lamson)  female   
9     1.0       0.0                          Artagaveytia, Mr. Ramon    male   

       age  sibsp  parch    ticket      fare    cabin embarked boat   body  \
0  29.0000    0.0    0.0     24160  211.3375       B5        S    2    NaN   
1   0.9167    1.0    2.0    113781  151.5500  C22 C26        S   11    NaN   
2   2.0000    1.0    2.0    113781  151.5500  C22 C26        S  NaN    NaN   
3  30.0000    1.0    2.0    113781  151.5500  C22 C26        S  NaN  135.0   
4  25.0000    1.0    2.0    113781  151.5500  C22 C26        S  NaN    NaN   
5  48.0000    0.0    0.0     19952   26.5500      E12        S    3    NaN   
6  63.0000    1.0    0.0     13502   77.9583       D7        S   10    NaN   
7  39.0000    0.0    0.0    112050    0.0000      A36        S  NaN    NaN   
8  53.0000    2.0    0.0     11769   51.4792     C101        S    D    NaN   
9  71.0000    0.0    0.0  PC 17609   49.5042      NaN        C  NaN   22.0   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON  
5                     New York, NY  
6                       Hudson, NY  
7                      Belfast, NI  
8              Bayside, Queens, NY  
9              Montevideo, Uruguay

### Zwróciła moja uwagę 4 osoby o nazwisku Allison, mieszkający w tych samych kajutach: ojciec (30 lat), matka (25 lat),2-letnia dziewczynka i prawie roczny chłopiec. Ktoś prawdopodobnie zaopiekował się chłopcem,  ponieważ tylko on z całej czwórki przeżył katastrofę. ####

### Przypatrzmy sie jeszcze szalupom ratunkowym:

In [12]:
df['boat'].value_counts()


boat
13         39
C          38
15         37
14         33
4          31
10         29
5          27
3          26
9          25
11         25
16         23
8          23
7          23
D          20
6          20
12         19
2          13
A          11
B           9
1           5
5 7         2
C D         2
13 15       2
5 9         1
8 10        1
13 15 B     1
15 16       1
Name: count, dtype: int64

### Pojawiaja sie podwojne liczby. Zakladam ze sa one bledne i je odrzucam. Pozostaje 20 szalup ratunkowych widoczna duza rozbieznosc ilosci ludzi w lodziach ratunkowych, w 13- 42 pasazerow, a w 1 - 5, 2- 13 . Jesliby przyjac ze lodzie ratunkowe byly zaladowywane kolejno tak jak sa ponumerowane, nasuwa sie wniosek, ze na poczatku ladowano tez do lodzi mniej pasazerow dopiero pozniej wypelniano lodzie wieksza iloscia ludzi. Wniosek jest taki, ze pomimo ze mogly pomiescic wiecej pasazerow, nie wykorzystaly pelnej ladownosci. Poniewaz jak wiadomo, statek mial byc niezatapialny, byc moze nawet zaloga nie wiedziala jaka jest calkowita pojemnosc szalup ratunkowych i sprawdzano to w trakcie wypelniania szalup. W mojej dalszej analizie przyjrzę się przeżywalności pasażerów Titanica.

# 2. Brakujace wartosci

In [13]:
df.isnull().sum() 

pclass          1
survived        1
name            1
sex             1
age           264
sibsp           1
parch           1
ticket          1
fare            2
cabin        1015
embarked        3
boat          824
body         1189
home.dest     565
dtype: int64

### Wszystkie wartosci, z wyjatkiem age, cabin, boat and body zastapie 'Unknown', age zastapie mediana w pozniejszym kroku. Celowo nie robie tego teraz, gdyz chce miec mozliwie najprawdziwsze dane do analizy

# 3. Eksplorujemy poszczegolne kolumny

## Na tym etapie chece przesledzic przezywalnosc i jakie okolicznosci na nia wplynely.

### Sprawdzmy, czy klasa biletu miala znaczenie jesli chodzi o przezywalnosc:

In [14]:
df_clean = df.dropna(subset=['pclass', 'survived'])

survival_by_class = df_clean.groupby('pclass')['survived'].sum()

total_by_class = df_clean['pclass'].value_counts().sort_index()

survival_rate_by_class = survival_by_class / total_by_class

fig, ax = plt.subplots()
ax.pie(survival_rate_by_class, labels=survival_rate_by_class.index, autopct='%1.1f%%', startangle=90)
ax.axis('equal')  
plt.title('Przezywalnosc wg. klasy biletu')

NameError: name 'plt' is not defined

In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt

sns. set_palette ( palette='Set1', n_colors = 3 ) # ustawmy ladna palete barw
df_clean = df.dropna(subset=['boat'])

boat_class_counts = df_clean.groupby(['boat', 'pclass']).size().unstack(fill_value=0)

fig, ax = plt.subplots(figsize=(12, 8))
boat_class_counts.plot(kind='bar', stacked=True, ax=ax)
ax.set_title('Liczba ludzi w szalupach ratunkowych wg. klasy biletu')
ax.set_xlabel('Numer szalupy ratunkowej')
ax.set_ylabel('Liczba ludzi')
ax.legend(title='Klasa biletu')

### Na wykresach widac, ze klasa biletu miala wplyw na przezywalnosc, gdzie pasazerowie 1 klasy mieli 47,5% szansy na przezycie, 2 klasy - 32,9%, a pasazerowie 3 klasy tylko 19,6%. Na wykresie 2 rowniez widac, ze lodzie byly poczatkowo wypelniane pasazerami 1 klasy, dopiero po wypelnieniu ok 1/3 lodzi, zaczeto je wypelniac 2 klasa, a na koncu 3 klasa. Widoczne jest tez, ze do szalup ratunkowych ladowano poczatkowo mniej ludzi, potem wiecej.

### Sprawdzmy teraz przezywalnosc jesli chodzi o plec:

In [ ]:
df_clean = df.dropna(subset=['survived', 'sex'])

plt.figure(figsize=(8, 6))
sns.countplot(data=df_clean, x='sex', hue='survived')
plt.title('Przezywalnosc wg. plci')
plt.xlabel('Plec')
plt.ylabel('Liczba')
plt.legend(title='Przezyli', labels=['Nie', 'Tak'])

### Wyraznie widac ze duzo wieksza szanse na przezycie mialy kobiety

### Sprawdzmy teraz przezywalnosc jesli chodzi o dzieci 1, 2, i 3 klasy:

In [ ]:
# Sprawdzmy teraz przezywalnosc jesli chodzi o dzieci 1, 2, i 3 klasy

# Dzieci obu plci o wieku mniejszym lub rownym 16 lat
children_df = df[df['age'] <= 16
]
grouped = children_df.groupby(['pclass', 'survived']).size().unstack()

fig, ax = plt.subplots(figsize=(8, 6))
grouped.plot(kind='bar', stacked=True, ax=ax)
ax.set_title('Przezywalnosc dzieci wg. klasy biletu')
ax.set_xlabel('Klasa biletu')
ax.set_ylabel('Liczba dzieci')
ax.legend(['Nie przezyli', 'Przezyli'], title='Przezywalnosc')

### Z powyzszego wykresu wynika, ze dzieci do 16 roku zycia i 1 i 2 klasy, mialy bardzo duza szanse przezycia. U dzieci w 3 klasie nie widac tej zaleznosci: wiecej dzieci w 3 klasie zginelo niz przezylo

# 4. Relacje miedzy kolumnami

### Przyjrzyjmy sie blizej przezywalnosci, jesli chodzi o plec i klase biletu:

In [ ]:

df_clean = df.dropna(subset=['pclass', 'survived', 'sex'])
pivot_table = df_clean.pivot_table(index='pclass', columns='sex', values='survived', aggfunc='mean')

plt.figure(figsize=(8, 6))
sns.heatmap(pivot_table, annot=True, cmap='coolwarm', cbar_kws={'label': 'Przezywalnosc'})
plt.title('Przezywalnosc wg. plci i klasy biletu')
plt.xlabel('Plec')
plt.ylabel('Klasa biletu')

### Widac ze najwieksza szanse na przezycie mialy kobiety 1 i 2 klasy, kobiety 3 klasy mialy juz mala szanse, a mezczyzni 1 klasy jeszcze mniejsza szanse na przezycie. Mezczyzni 2 i 3 klasy praktycznie nie mieli szansy przezycia

# 5. Poszukiwanie wartosci odstajacych

### Narysujmy wykresy ceny biletu i przezywalnosci:

In [ ]:
survived = df[df['survived'] == 1]
not_survived = df[df['survived'] == 0]

fig, axes = plt.subplots(1, 2, figsize=(12, 6))

sns.boxplot(ax=axes[0], y=not_survived['fare'])
axes[0].set_title('Cena biletu pasazerow ktorzy nie przezyli')
axes[0].set_ylabel('Fare')

sns.boxplot(ax=axes[1], y=survived['fare'])
axes[1].set_title('Cena biletu pasazerow ktorzy przezyli')
axes[1].set_ylabel('Cena')

### Przyjmujac, ze cena zalezala od klasy biletu i pamietajac ze pierwsza klasa byla wyraznie preferowana jesli chodzi o dostep do lodzi ratunkowych,przykuwa uwage duza liczba wartosci odstajacych u pasazerow ktorzy nie przezyli. Sa to prawdopodobnie mezczyzni 1 klasy, ktorzy odstapili miejsca w szalupach ratunkowych kobietom i dzieciom
    

# 6. Transformacja danych / naprawa bledow

### Wszystkie wartosci, z wyjatkiem age, cabin, boat and body zastapie 'Unknown', age zastapie mediana

In [ ]:
df = df.fillna({
    'age': df['age'].median(),
    'pclass': 'Unknown',
    'survived': 'Unknown',
    'name': 'Unknown',
    'sex': 'Unknown',
    'sibsp': 'Unknown',
    'parch': 'Unknown',
    'ticket': 'Unknown',
    'fare': 'Unknown',
    'embarked': 'Unknown',
    'home.dest': 'Unknown',
  
})
df

# **Podsumowanie**

### 1.    RMS Titanic podróżowało **1310 pasażerów**. Statek mial 890 osób z zalogi.  Wielu pasażerów podróżowało statkiem Titanic do Ameryki w celach osiedleńczych, ale tez niektórzy bardziej zamożni podróżowali w celach wycieczkowych czy biznesowych. Na statku były **3 klasy** pasażerów, 186 kajut było rozdysponowanych pomiędzy pasażerów 1 i 2 klasy. **Najmłodszy pasażer mial 1 miesiąc, a najstarszy 80 lat**.

### 2. Widoczny **duży dramatyzm** danych, np 4- osobowa **rodzina Allisonow**: ojciec (30 lat), matka (25 lat),2-letnia dziewczynka i prawie roczny chłopiec. Ktoś prawdopodobnie zaopiekował się chłopcem, ponieważ tylko on z całej czwórki przeżył katastrofę.


### 3. Przyczyna smierci pasazerow Titanica byla zbyt **mala liczba 20 szalup ratunkowych**. Zwraca tez uwage **duza rozbieznosc liczby osob w lodziach ratunkowych**. Widać, ze na poczatku ladowano do lodzi mniej pasazerow dopiero pozniej wypelniano lodzie wieksza iloscia ludzi. Wniosek jest taki, ze pomimo ze mogly pomiescic wiecej pasazerow, **nie wykorzystaly pelnej ladownosci**. Poniewaz jak wiadomo, statek mial byc niezatapialny, byc moze nawet zaloga nie wiedziala jaka jest calkowita pojemnosc szalup ratunkowych i sprawdzano to w trakcie wypelniania szalup. 

### 4. Z wykresów wynika, ze **klasa biletu miala wplyw na przezywalnosc**, gdzie pasazerowie 1 klasy mieli 47,5% szansy na przezycie, 2 klasy - 32,9%, a pasazerowie 3 klasy tylko 19,6%. Lodzie byly poczatkowo wypelniane pasazerami 1 klasy, dopiero po wypelnieniu ok 1/3 lodzi, zaczeto je wypelniac 2 klasa, a na koncu 3 klasa. 

### 5. Widac ze **najwieksza szanse na przezycie** mialy **kobiety i dzieci 1 i 2 klasy** - to ich przepuszczano przodem do szalup ratunkowych i one miały praktycznie całkowite szanse na przetrwanie. Kobiety i dzieci 3 klasy mialy juz mala szanse, a mezczyzni 1 klasy jeszcze mniejsza szanse na przezycie. **Mezczyzni 2 i 3 klasy praktycznie nie mieli szansy przetrwania**.

### 6. Jest 1 tajemniczy pasażer (1309) o którym nic nie wiadomo. Być może to on posłużył jako inspiracja dla charakteru Jacka, z filmu Jamesa Camerona?

![Infografika rozkladu lodzi ratunkowych](https://upload.wikimedia.org/wikipedia/commons/thumb/4/43/RMS_Titanic_boat_deck_plan_EN.svg/1176px-RMS_Titanic_boat_deck_plan_EN.svg.png)

Infografika układu łodzi ratunkowych Titanica. Kropki oznaczają liczbę osób na łodzi i maksymalną pojemność.